presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -187,13 +194,44 @@ private boolean isLowHighSupportedForType(Type type)
                 || type.equals(BIGINT)
                 || type.equals(REAL)
                 || type.equals(DOUBLE)
-                || type.equals(DATE)
-                || type.equals(TIMESTAMP)) {
+                || type.equals(DATE)) {
             return true;
         }
         return false;
     }
 
+    public static com.facebook.presto.spi.statistics.Range createPrestoRange(Type type, Object min, Object max)
+    {
+        return new com.facebook.presto.spi.statistics.Range(convertPrestoTypeValueToDouble(type, min), convertPrestoTypeValueToDouble(type, max));
+    }
+
+    private static double convertPrestoTypeValueToDouble(Type type, Object value)
```

Comment:
"ToDouble" doesn't convey the intended semantics; `convertPrestoValueToStatsRepresentation`